Data/ByteString/Streaming/Char8.hs

```
@@ -386,28 +390,23 @@ repeat = R.repeat . c2w
 -- | 'cycle' ties a finite ByteString into a circular one, or equivalently,
 -- the infinite repetition of the original ByteString.
 --
--- | /O(n)/ The 'unfoldr' function is analogous to the Stream \'unfoldr\'.
--- 'unfoldr' builds a ByteString from a seed value.  The function takes
--- the element and returns 'Nothing' if it is done producing the
--- ByteString or returns 'Just' @(a,b)@, in which case, @a@ is a
--- prepending to the ByteString and @b@ is used as the next element in a
--- recursive call.
+-- | /O(n)/ The 'unfoldM' function is analogous to the Stream \'unfoldr\'.
+-- 'unfoldM' builds a ByteString from a seed value. The function takes the
+-- element and returns 'Nothing' if it is done producing the ByteString or
+-- returns 'Just' @(a,b)@, in which case, @a@ is a prepending to the ByteString
+-- and @b@ is used as the next element in a recursive call.
```

The name here is unfortunate IMHO (but not much can be done about that now). I'd expect `unfoldM` to take a _monadic_ step (as do similarly named functions in other packages).

That said, the description should be better I think. The key difference from the `unfoldr` function below, is that the final return value `r` here is always unit, and so the step is a `Maybe` rather than an `Either r`. Otherwise, it is a vanilla unfold.  [ So I guess what's "M"onadic here, in various use-cases, is not the unfold step, but perhaps rather the value of `r`, which may be a monadic continuation for the outer stream. ]

It is however probably worth mentioning both here and below, that characters are silently truncated to 8 bits.

---

I noticed that too while fixing the docs. Why `M` was chosen here I'm not sure.